src/services/comment.js

```diff
@@ -4,11 +4,19 @@ const commentService = {
   addComment: async (data) => {
     const { text, author, authorRole, cooperationId } = data
 
-    return await Comment.create({ author, cooperation: cooperationId, text, authorRole })
+    const newComment = await Comment.create({ author, cooperation: cooperationId, text, authorRole })
+
+    return await Comment.findById({ _id: newComment._id })
+      .populate('author', 'firstName lastName')
+      .select('-authorRole')
+      .exec()
   },
 
   getComments: async (cooperationId, userId) => {
-    return await Comment.find({ cooperation: cooperationId, author: userId }).exec()
+    return await Comment.find({ cooperation: cooperationId, author: userId })
```

Comment:
Have cases with non-existent `cooperationId` and/or `userId` been checked?

Reply:
`cooperationId` is part of the route and the `userId` should be in the token, so the outer code handles these specific cases.